### ISSUE TITLE
Incompatibility with Next.js 15

### ISSUE THREAD

**Issue by Niklas-Mezynski** [ethanniser/next-typesafe-url#107]
The new Next.js 15 release brought some breaking changes to the Async Request APIs (including search params)
https://nextjs.org/blog/next-15#async-request-apis-breaking-change

This currently breaks the library.

I currently don't have time to fix it myself and I cannot estimate the effort to migrate the library. Maybe someone else can help here

**Comment by ethanniser** [comment]
Aware of this- will look sometime this week/weekend

**Comment by ethanniser** [comment]
please try out the pkr.pr.new snapshot in the draft pr !

**Comment by Niklas-Mezynski** [comment]
Thanks a lot, looking good so far. I just upgraded my package to `"next-typesafe-url": "https://pkg.pr.new/ethanniser/next-typesafe-url@110"` and it seems to work as a drop in replacement. I only had to change some of my types from smth like `PageProps["searchParams"]` to `Awaited<PageProps["searchParams"]>`. As far as I could test, the functionality works as expected. I am using the search params API and type safe url construction with `$path`
